api/nameguard/nameguard.py

```diff
@@ -350,6 +351,14 @@ async def secure_primary_name(self, address: str, network_name: str) -> SecurePr
                                        primary_name_status=status,
                                        nameguard_result=nameguard_result)
 
+    async def fake_eth_name_check_fields(self, network_name, contract_address, token_id, investigated_fields: dict[str,str]) -> FakeEthNameCheckResult:
+        contract_address = contract_address.lower()
+        
+        # title = investigated_fields['title']  # mandatory if ENS contract; ENS name
+        
+        #TODO: return only impersonating fields
```

Comment:
@lightwalker-eth For now, we have been returning all fields obtained from Alchemy. It doesn't make sense in the new endpoint because we would have to return the same dict as provided by client. We can return only fields which are impersonating ETH name. Do we want different response for these endpoints?

Reply:
So we will return only fields which are faking ENS names. However, we break the loop if the first is faking field encountered so there always be at most only one key in response.

Reply:
Sure, we can, but ens_cure is quite expensive (n^2).